src/odil/DataSet.cpp

```diff
@@ -235,7 +235,7 @@ ::operator[](Tag const & tag)
     ElementMap::iterator it = this->_elements.find(tag);
     if(it == this->_elements.end())
     {
-        throw Exception("No such element");
+        throw Exception("No such element "+ tag.get_name());
```

Comment:
`Tag.get_name` may throw (e.g. for private tags): the safest way would be to use `operator std::string()` instead.